rfcs/0075-declarative-wrappers.md

```diff
@@ -0,0 +1,408 @@
+manual](https://github.com/NixOS/nixpkgs/blob/2df97e4b0ab73f0087af2e6f33e694140150db1b/doc/languages-frameworks/gnome.section.md#L120-L166)
+
+Declarative wrappers will deprecate the usage of our shell based hooks and will
+wrap all executables automatically according to their needs, without requiring
+the contributor a lot of knowledge of the wrapping system. Also, double
+wrappings will become a problem of the past.
+
+- [issue 86369](https://github.com/NixOS/nixpkgs/issues/86369) 
+
+@ttuegel with declarative wrappers, we can symlink all qt plugins into 1
+directory and wrap the executable with only 1 `QT_PLUGIN_PATH` in their
+environment, which should decrease the plugin load of every qt package.
+
+## Issues _possibly_ fixable by declarative wrappers (?)
+
+- [pull 61213](https://github.com/NixOS/nixpkgs/pull/61213) 
+
+I'm not sure what's the issue there. But, I'm sure that a declarative, Nix
+based builder of a Python environment, even if this environment is used only
+for a build, should make it easier to control and alter it's e.g `$PATH`.
+
+- [issue 49132](https://github.com/NixOS/nixpkgs/issues/49132)
+- [issue 54278](https://github.com/NixOS/nixpkgs/issues/54278)
+- [issue 39493](https://github.com/NixOS/nixpkgs/issues/39493)
+
+`GDK_PIXBUF` compatibilities? I haven't investigated them to the details, so feel
+free @jtojnar to review me and tell me that declarative wrappers won't help.
+
+## Unreported issues (AFAIK)
+
+Issues that bother me personally, but I haven't bothered to open an issue since
+I doubt it would be feasible to fix with our current wrapping ecosystem, excuse
+my pessimism `wrap{G,Qt}AppsHook` authors.
+
+`kdeconnect` has `kdoctools` in it's closure because it's wrapper has
+`kdoctools` due to it picked by `wrapQtAppsHook`:
+
+```
+$ nix why-depends -f. kdeconnect kdoctools
+/nix/store/sh42k6cz4j48br4cxi2qn173rys4japp-kdeconnect-1.3.5
+╚═══bin/kdeconnect-cli: …xport XDG_DATA_DIRS='/nix/store/m16681i5dhhkhszi9w42ir037jvbnab9-kdoctools-5.71.0/share'${XDG_DA…
+    => /nix/store/m16681i5dhhkhszi9w42ir037jvbnab9-kdoctools-5.71.0
+```
+
+A similar issue is with `kconfigwidgets.dev` and `kdeconnect`:
+
+```
+$ nix why-depends -f. kdeconnect kdeframeworks.kconfigwidgets.dev
+ /nix/store/sh42k6cz4j48br4cxi2qn173rys4japp-kdeconnect-1.3.5
+ ╚═══bin/kdeconnect-cli: …port QT_PLUGIN_PATH='/nix/store/qssjj6ki7jiskw2kfygvfiy8fxrclwrl-kconfigwidgets-5.71.0-dev/lib/q…
+     => /nix/store/qssjj6ki7jiskw2kfygvfiy8fxrclwrl-kconfigwidgets-5.71.0-dev
+```
+
+Also similar (but possibly fixable by moving `gobject-introspection` to a
+different inputs list?):
+ 
+```
+$ nix why-depends -f. beets gobject-introspection.dev
+/nix/store/93lfrhm8vp17m8ziqi8vp6v4cff67wkb-beets-1.4.9
+╚═══bin/beet: …-expat-2.2.8-dev/bin:/nix/store/y3ym76wrak3300vsjyf3klr52cnzmxwd-gobject-introspection-1.64.1-de…
+    => /nix/store/y3ym76wrak3300vsjyf3klr52cnzmxwd-gobject-introspection-1.64.1-dev
+```
+
+## Other issues
```

Comment:
More related issues with wrappers https://github.com/NixOS/nixpkgs/issues/160923 and the ones linked at https://github.com/NixOS/nixpkgs/issues/160923#issuecomment-1046067348